webapp/components/features/FollowList/FollowList.spec.js

```diff
@@ -0,0 +1,150 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followingCount)
```

Comment:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-04-15T13:53:23Z" title="Wednesday, April 15th 2020, 3:53:23 pm +02:00">Apr 15, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,150 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.connections).toEqual(user.following)
```


It is better to avoid `.vm.` constructions when testing, because this way the test is against the internal state of the component and not against what is actually rendered.
May be something like:
```javascript
expect(wrapper.findAll('.user-teaser')).toHaveLength(user.following.length)
```

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-15T21:37:44Z" title="Wednesday, April 15th 2020, 11:37:44 pm +02:00">Apr 15, 2020</time>_

---


ee3373117